### PR TITLE
Connected to database, Added classes for global exception handling and class for swagger config

### PR DESCRIPTION
**Pull request by BeattLau**
Sorry, didn't mean to push everything(classes other than those in the title are just empty and for me to organise structure.
Also sure that all classes are not complete but I just wanted to push what I've achieved yesterday and today.

### CODE REVIEW

File: .idea/.gitignore
Comment (faruka-startsteps) [inline .idea/.gitignore:1]: .idea should be in git ignore.

File: src/main/java/ExceptionHandler/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+@ControllerAdvice
+public class GlobalExceptionHandler {
+    @ExceptionHandler({UserNotFoundException.class})
+    public ResponseEntity<Object> handleUserNotFoundException(UserNotFoundException exception) {
+        return ResponseEntity
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
```
Comment (faruka-startsteps) [inline src/main/java/ExceptionHandler/GlobalExceptionHandler.java]: server error is not a good response if the user is not found. With server error you are implying that something wrong happened in the backend, but nothing did, you just can't find the user

File: src/main/java/ExceptionHandler/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                .body(exception.getMessage());
+    }
+    @ExceptionHandler({UserAlreadyExistsException.class})
+    public ResponseEntity<Object> handleUserAlreadyExistsException(UserAlreadyExistsException exception) {
+        return ResponseEntity
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
```
Comment (faruka-startsteps) [inline src/main/java/ExceptionHandler/GlobalExceptionHandler.java]: similar to above

File: src/main/java/ExceptionHandler/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                .body(exception.getMessage());
+    }
+    @ExceptionHandler({RuntimeException.class})
+    public ResponseEntity<Object> handleRuntimeException(RuntimeException exception) {
+        return ResponseEntity
+                .status(HttpStatus.INTERNAL_SERVER_ERROR)
```
Comment (faruka-startsteps) [inline src/main/java/ExceptionHandler/GlobalExceptionHandler.java:24]: This is appropriate 😄

File: src/main/java/User/Model/Roles.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package User.Model;
+
+public enum Roles {
+        ROLE_CUSTOMER,
+        ROLE_ADMIN
+    }
```
Comment (faruka-startsteps) [inline src/main/java/User/Model/Roles.java]: extra space

File: src/main/java/User/Model/Roles.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
```
Comment (faruka-startsteps) [inline src/main/java/User/Model/Roles.java]: remove line

File: src/main/java/User/Dto/UserRegistrationDto.java
Comment (faruka-startsteps) [inline src/main/java/User/Dto/UserRegistrationDto.java:1]: remove if not needed

File: src/main/java/User/Dto/UserLoginDto.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package User.Dto;
+
+public class UserLoginDto {
```
Comment (faruka-startsteps) [inline src/main/java/User/Dto/UserLoginDto.java:3]: remove if not needed

File: src/main/java/User/Controller/UserController.java [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package User.Controller;
+public class UserController{
```
Comment (faruka-startsteps) [inline src/main/java/User/Controller/UserController.java:2]: remove if not needed

File: src/main/java/User/Dto/UserRegistrationDto.java [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package User.Dto;
+
+public class UserRegistrationDto {
```
Comment (faruka-startsteps) [inline src/main/java/User/Dto/UserRegistrationDto.java:3]: remove if not needed

File: src/main/java/User/Model/User.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
+package User.Model;
+
+import jakarta.persistence.*;
+import java.util.Set;
+@Entity
```
Comment (faruka-startsteps) [inline src/main/java/User/Model/User.java:5]: add empty line before this one

File: src/main/java/User/Service/UserService.java [hunk 10]
```diff
@@ -0,0 +1,6 @@
+package User.Service;
+
+
+
+public interface UserService {
```
Comment (faruka-startsteps) [inline src/main/java/User/Service/UserService.java:5]: remove if not needed

File: src/main/java/User/Service/UserService.java [hunk 11]
```diff
@@ -0,0 +1,6 @@
+package User.Service;
+
+
```
Comment (faruka-startsteps) [inline src/main/java/User/Service/UserService.java:3]: remove two lines

File: src/main/java/User/Service/UserServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,9 @@
+package User.Service;
+
+
```
Comment (faruka-startsteps) [inline src/main/java/User/Service/UserServiceImpl.java:3]: remove three lines

File: src/main/java/User/Service/UserServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+
+
+public class UserServiceImpl {
```
Comment (faruka-startsteps) [inline src/main/java/User/Service/UserServiceImpl.java:6]: remove if not needed

File: src/main/java/eCommerce/LoggingConfig.java [hunk 14]
```diff
@@ -0,0 +1,32 @@
+package eCommerce;
+
+import java.io.IOException;
+import java.util.logging.*;
+
+
```
Comment (faruka-startsteps) [inline src/main/java/eCommerce/LoggingConfig.java]: extra space

File: src/main/java/eCommerce/eCommerceFinalProjectApplication.java [hunk 15]
```diff
@@ -0,0 +1,15 @@
+package eCommerce;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+
+@SpringBootApplication
+
```
Comment (faruka-startsteps) [inline src/main/java/eCommerce/eCommerceFinalProjectApplication.java:8]: extra space

**Review by faruka-startsteps — CHANGES_REQUESTED** [review]

**Comment by faruka-startsteps** [comment]
please be more descriptive with branch naming. `dev` doesn't mean anything to the person looking at it

**Comment by faruka-startsteps** [comment]
Ideally we should have one branch per ticket, and the naming should follow a structure. So if you are working on [this ticket](https://beattlau.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-34) the name of the branch would be SCRUM-34-swagger

**Comment by BeattLau** [comment]
Update classes after being reviewed

**Pull request closed without merging**